lib.sh

```diff
@@ -222,9 +223,9 @@ domain_check_200() {
     else
         print_text_in_color "$IRed" "DNS lookup failed with nslookup."
         print_text_in_color "$IRed" "Please check your DNS settings! Maybe the domain isn't propagated?"
-	print_text_in_color "$ICyan" "Please check https://www.whatsmydns.net/#A/${1} if the IP seems correct."
+        print_text_in_color "$ICyan" "Please check https://www.whatsmydns.net/#A/${1} if the IP seems correct."
         nslookup "${1}" $DNS1
-        return 1
+        exit 1
```

Comment:
Are there any special reason you changed this?

Usually we send return instead of exit to answer a question in a function that is used. Exit will exit the whole thing, and that's not what we want if the user are running the first setup script for example.

Reply:
should the process stop immediately， but when a domain can't be accessed。I make some tests that I type the error domain， some strange things appeared.

Reply:
Hmm, I don't understand?

Reply:
> Hmm, I don't understand?

the function, domain_check_200(), is used to check if a domain can be accessed by using nslookup. when the command failed, it indicates that user types an error domain. So I think the process should be stopped by "exit" instead of "return".  if using return, the process will continue to execute   "dig....", and the screen will show: "DNS seems correct when checking with dig!".

Reply:
> Hmm, I don't understand?
The reason I said above is incorrect。My real reason is as follows：
The function, domain_check_200(), is used to check if a domain can be accessed by using `nslookup`. when the command failed, the function will be return directly don't execute the following code  that is :
```
 DIG="$(dig +short "${1}" @resolver1.opendns.com)"                                                                                                                                                                                                                 
 if [[ "$DIG" = "$WANIP4" ]]
 then
        print_text_in_color "$IGreen" "DNS seems correct when checking with dig!"
 elif [[ "$DIG" != "$WANIP4" ]]
... ...
.... ...
```

This causes the script that called the function to continue executing，and don‘t know the domain name is wrong.

Reply:
The reason why I want it to continue is that the checks are independent. Sometimes it works with DIG, and not with NSLOOKUP. That's why the two exist.

I also think that the whole thing in generate-cert.sh should be in `lib.sh` as there's where all the functions exists. Not a separate script. Please change.

Reply:
I will do another test using original script

Reply:
I have done the test,  Using the original script, no bad thing happen. when `nslookup` failed, the function return. Then,  the script will stop because of `check_open_port`.  

Now, I changed the behavior of `check_open_port`, if one types "yes" to answer the question which is "Are you sure that your port 80/443 is open ".  the script will continue to execute the following code. 

That's why I did that.

Reply:
Well, either the port is closed or it's open. Why add a question that's asking if it's really open?

The check happens externally anyway, and the TLS cert generation will fail if the port is closed. 

Do you see the logic here?

Reply:
When the ISP banned the port detecting/scan from external, the User can't active SSL. In fact, their port is open,  but no response can be received from https://ports.yougetsignal.com/check-port.php.

Reply:
OK, add a `msg_box` instead, explaining that. Something like:

```
msg_box "It seems like the port [port] is closed. This could be because your ISP has blocked the port, or that the port isn't open.

If you are 100% sure the port is open you can now choose to continue. There are no guarantees that it will work anyway though, since Let's Encrypt depend on that the port is open and accessible from outside your network."

ask_yes_or_no [Are you 100% sure the port is open?] Yes/No
```

Thanks!